html/infrastructure/safe-passing-of-structured-data/structuredclone_0.html

```diff
@@ -367,6 +367,154 @@
+                    worker.postMessage(error);
+                });
+            },
+            function() {
+                var t = async_test("URIError objects can be cloned");
+                t.id = 33;
+                worker.onmessage = t.step_func_done(function(e) {
+                    assert_equals(Object.getPrototypeOf(e.data), URIError.prototype, "Checking prototype");
+                    assert_equals(e.data.constructor, URIError, "Checking constructor");
+                    assert_equals(e.data.name, "URIError", "Checking name");
+                    assert_equals(e.data.message, "some message", "Checking message");
+                    assert_equals(e.data.foo, undefined, "Checking custom property");
+                });
+                t.step(function() {
+                    const error = URIError("some message");
+                    error.foo = "bar";
+                    worker.postMessage(error);
+                });
+            },
+            function() {
+                var t = async_test("Cloning a modified Error");
+                t.id = 34;
+                worker.onmessage = t.step_func_done(function(e) {
+                    assert_equals(Object.getPrototypeOf(e.data), SyntaxError.prototype, "Checking prototype");
+                    assert_equals(e.data.constructor, SyntaxError, "Checking constructor");
+                    assert_equals(e.data.name, "SyntaxError", "Checking name");
+                    assert_equals(e.data.message, "another message", "Checking message");
+                    assert_equals(e.data.foo, undefined, "Checking custom property");
+                });
+                t.step(function() {
+                    const error = URIError("some message");
+                    Object.setPrototypeOf(error, SyntaxError.prototype);
+                    error.message = {toString: () => "another message" }
+                    error.constructor = RangeError;
+                    error.name = "TypeError";
+                    error.foo = "bar";
+                    worker.postMessage(error);
+                });
+            },
+            function() {
+                var t = async_test("DOMException objects can be cloned");
+                t.id = 35;
+                worker.onmessage = t.step_func_done(function(e) {
+                    assert_equals(Object.getPrototypeOf(e.data), DOMException.prototype, "Checking prototype");
+                    assert_equals(e.data.constructor, DOMException, "Checking constructor");
+                    assert_equals(e.data.name, "IndexSizeError", "Checking name");
+                    assert_equals(e.data.message, "some message", "Checking message");
+                    assert_equals(e.data.code, 1, "Checking code");
+                    assert_equals(e.data.foo, undefined, "Checking custom property");
+                });
+                t.step(function() {
+                    const error = new DOMException("some message", "IndexSizeError");
```

Comment:
Can we test also on some error thrown by DOM functions?

Reply:
Done.